### PR TITLE
Split into package for library and package for executable

### PR DESCRIPTION
**Pull request by mmhat**
This PR splits the `grace` package into two:

- The `grace-core` package which is meant to be used programmatically, e.g. for tooling or embedded interpreters.
- The `grace` package which is the home of the grace executable.

### CODE REVIEW

**Comment by Gabriel439** [comment]
I think this is a great idea!  My main comments so far are:

* I think `Grace.REPL` should be part of the executable package
* The executable package doesn't need a `doctest` suite
* I would like to keep the current `haskell.yml` for CI right now, because it's much simpler

**Comment by mmhat** [comment]
@Gabriel439 This one is ready for review; Feel free to merge it if there is nothing to object to :slightly_smiling_face:

**Comment by Gabriel439** [comment]
Thank you for doing this! 🙂

**Comment by Gabriel439** [comment]
@mmhat: Actually, I wanted to ask you if it would be alright to go back to a single consolidated package?

I'm sorry to ask this after the work you put into this change, but the main reason I'm requesting this is because I'm running into the project boundary pretty frequently in the course of development where I need to make changes that span both packages.  In particular, both `ghcid` and Nix do not support fast incremental development that span a package boundary.

I also tried using `cabal v2-build` (without Nix) instead to do cross-package development, but ran into a failure getting that to work, too.

**Comment by mmhat** [comment]
@Gabriel439 Yes, I have no problem with that! It seemed to be a good idea back then to reduce the dependency footprint if one wants to use the core library component only. But I made the split under the assumption that the packages will end up on Hackage and without that it is IMHO not really important to have two packages floating around.
If requirements change, let's change the project, too!
TLDR; Happy merging :smile:
